File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        i = 1
        max_profit = 0
        while i < len(prices):
            max_profit = max(prices[i] - prices[left], max_profit)
            if prices[i] < prices[left]:
                left = i 
            i += 1
        return max_profit

File: Data Structures & Algorithms/trapping-rain-water/submission-2.java
class Solution {
    public int trap(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int leftMax = height[0];
        int rightMax = height[right];
        int sol = 0;

        while (left < right){
            if (leftMax < rightMax){
                left++;
                leftMax = Math.max(height[left], leftMax);
                sol += leftMax - height[left];
            } else {
                right--;
                rightMax = Math.max(height[right], rightMax);
                sol += rightMax - height[right];
            }
        }

        return sol;
    }
}
